### PR TITLE
AEA-3247 Connect to spine for status endpoint

### PR DESCRIPTION
**Pull request by anthony-nhs**

### CODE REVIEW

File: packages/spineClient/src/live-spine-client.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import {ClientRequest, SpineResponse} from "./models/spine"
```
Comment (originalphil) [inline packages/spineClient/src/live-spine-client.ts:1]: Are the tests for this to be added when it's returning a non-400 response?

File: Makefile [hunk 2]
```diff
@@ -88,25 +89,54 @@ sam-deploy-package: guard-artifact_bucket guard-artifact_bucket_prefix guard-sta
...
+lint: lint-node lint-go
 
-test:
+test: compile
 	npm run test --workspace packages/capabilityStatement
 	npm run test --workspace packages/getMyPrescriptions
 	npm run test --workspace packages/middleware
 	npm run test --workspace packages/sandbox
 	npm run test --workspace packages/statusLambda
+	npm run test --workspace packages/spineClient
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply Makefile:121]: there are not any golang tests. The go program just gets a secret and outputs it using standard aws/golang libraries so not really sure what could be tested

File: packages/getSecretLayer/src/go-retrieve-secret.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+	// output the secret value
+	fmt.Printf(*result.SecretString)
+}
+
+func getCommandParams() {
+	// Setup command line args
+	flag.StringVar(&region, "r", DEFAULT_REGION, "The Amazon Region to use")
+	flag.StringVar(&secretArn, "s", "", "The ARN for the secret to access")
+	flag.StringVar(&roleArn, "a", "", "The ARN for the role to assume for Secret Access")
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply packages/getSecretLayer/src/go-retrieve-secret.go:81]: yep - removed it

File: packages/getSecretLayer/src/go-retrieve-secret.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+	// Verify that the correct number of args were supplied
+	if len(region) == 0 || len(secretArn) == 0 {
+		flag.PrintDefaults()
+		panic("You must supply a region and secret ARN.  -r REGION -s SECRET-ARN [-a ARN for ROLE -t TIMEOUT IN MILLISECONDS -n SESSION NAME]")
+	}
+}
+
+// This function will attempt to assume the supplied role and return either an error or the assumed role
+func AttemptAssumeRole(ctx context.Context, cfg aws.Config) (*sts.AssumeRoleOutput, error) {
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply packages/getSecretLayer/src/go-retrieve-secret.go:96]: yep - removed it

File: packages/spineClient/src/live-spine-client.ts (hunk 1 above)
Comment (anthony-nhs) [reply packages/spineClient/src/live-spine-client.ts:1]: yes

**Review by originalphil — APPROVED** [review]

**Pull request merged**